doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+.. _sensor-threat:
+
+Sensor Device Threat Model
+##########################
+
+This document describes a threat model for an IoT sensor device. This
+device contains a sensor of some type (for example temperature, or a
+pressure in a pipe), which sends this data to an SoC running a
+microcontroller. This microcontroller connects to a cloud service, and
+relays this sensor data to this service. The cloud service is also able
+to send configuration data to the device, as well as software update
+images. A general diagram can be seen in Figure 1:
+
+.. figure:: media/sensor-model.svg
+
+   Figure 1. Sensor General Diagram
+
+In this sensor device, the sensor connects with the SoC via an SPI bus,
+and the SoC has a network interface that it uses to communicate with the
+cloud service. The particulars of these interfaces can impact the threat
+model in unexpected ways, and variants on this will need to be
+considered (for example, using a separate network interface SoC
+connected via some type of bus).
+
+This model also focuses on communicating via the MQTT-over-TLS protocol,
+as this seems to be in wide use [1]_.
+
+Assets
+======
+
+One aspect of the threat model to consider are assets involved in the
+operation of the device. The following list enumerates the assets
+included in this model:
+
+1. **The bootloader**. This is a small code/data image contained in
+   on-device flash that is the first code to run. In order to establish
+   a root of trust, this image must be immutable. This model assumes
+   that the SoC provides a mechanism to protect a region of the flash
+   from future writes, and that this will be done after this image is
```

Comment:
@d3zd3z I wonder, here: shouldn't there be a _looser_ requirement/assumption for _read_ protection, too (besides WRITE)? E.g. to mitigate risks of reverse-engineering the image via reading it.

Reply:
The bootloader is pretty hard to read-protect, since the CPU needs to read it in order to run it. That permission could be removed later. However, the bootloader is open source, and there isn't all that much to be gained by reverse engineering it. The keys are public keys, or hashes of public keys, and, as such, are publicly available.

The difference is that the write protection _must_ be provided at a hardware level, and not just via register configurations that are written by software. Any read protection would just be software settings that come later.